### PR TITLE
feat: Feature flag for pure Rust crypto

### PR DESCRIPTION
**Pull request by maciejhirsz**
@tomusdrw this is really dirty atm, before I get to clean this up I'd like to know if it actually makes sense to do it here, or wether we should push it upstream to `parity-crypto` or make it its own crate?

### CODE REVIEW

**Review by tomusdrw — COMMENTED** [review]
Looks ok, remember to update `.travis.yml` to compile using new feature.

Regarding options:
1. I think the best would be to put it as a feature to `parity-crypto`, but it might actually contain more things then we need and introducing it might be too dificult.
2. Separate crate seems like a good option too, we can either host it in this repo or completely outside if you prefer.

**Comment by maciejhirsz** [comment]
Aight, went with making it a drop-in replacement crate, updated Travis scripts.

edit: Also, will need to publish the crate and then change the dep in main `Cargo.toml` before publishing that.

File: ethsign-crypto/src/aes.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+use std::fmt;
+use aes_ctr::Aes128Ctr;
+use aes_ctr::stream_cipher::generic_array::GenericArray;
+use aes_ctr::stream_cipher::{NewStreamCipher, SyncStreamCipher};
+
+#[derive(Debug)]
+pub enum SymmError {
```
Comment (tomusdrw) [inline ethsign-crypto/src/aes.rs:10]: Docs?

File: ethsign-crypto/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
+//! Pure Rust drop-in replacement for the `parity-crypto` crate.
+
```
Comment (tomusdrw) [inline ethsign-crypto/src/lib.rs:2]:
```suggestion
`#![warn(missing_docs)]`
```

**Review by tomusdrw — APPROVED** [review]
looks good except for missing docs

**Comment by maciejhirsz** [comment]
@tomusdrw added some comments. I'm not feeling super comfortable documenting the whole thing as I actually don't know what all the parts are for - just porting them from `parity-crypto` 1:1, which itself lacks documentation.

I'll make an issue there and we can work it out from there on.

**Comment by tomusdrw** [comment]
Fair enough!

**Pull request merged**